### PR TITLE
ggplot2, heatmap.2, reshape2, and tSNE

### PR DESCRIPTION
**Pull request by MoHeydarian**
Migrating ggplot2, heatmap.2, reshape2, and tSNE wrappers from the test
toolshed.

The ggplot2 wrappers correspond to the geom_point, geom_freqpoly, and
geom_violin functions (scatterplot, histogram, and violin plots, res.)
of the ggplot2 package in R.

The heatmap2 wrapper uses the heatmap.2 function of R.

The reshape2 wrappers correspond to the melt and dcast functions of the
reshape2 package in R.

The tSNE uses the Rtsne function in R.

### CODE REVIEW

File: tools/ggplot2/ggplot_histogram.xml [hunk 1]
```diff
@@ -0,0 +1,192 @@
+<tool id="ggplot_histogram" name="Histogram w ggplot2" version="0.1.3">
+    <requirements>
+        <requirement type="package">r-getopt</requirement>
+        <requirement type="package">r-ggplot2</requirement>
+        <requirement type="package">r-reshape2</requirement>
+    </requirements>
+    <stdio>
+        <exit_code range="1:" />
```
Comment (bgruening) [inline tools/ggplot2/ggplot_histogram.xml]: You can remove the stdio tag and use `<command detect_errors="exit_code">`

File: tools/ggplot2/ggplot_histogram.xml [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    </stdio>
+    <command><![CDATA[
+        Rscript ${__tool_directory__}/ggplotscripthistogram.R --input "$input1" --title "$title"
```
Comment (bgruening) [inline tools/ggplot2/ggplot_histogram.xml]: Can you put ${__tool_directory__}/ggplotscripthistogram.R in single quotes, please.
The other inputs, outputs and text parameters should also be single quotes.

File: tools/ggplot2/ggplot_histogram.xml [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+            --houtputdim ${dimensions.houtputdim}
+        #else:
+           --woutputdim=7
+           --houtputdim=7
+        #end if
+
+    ]]></command>
+    <inputs>
+        <param name="input1" type="data" format="tabular" label="Input should have column headers - these will be the columns that are plotted"/>
+        <param name="title" size="30" type="text"  format="txt"/>
```
Comment (bgruening) [inline tools/ggplot2/ggplot_histogram.xml]: size="" can be removed from all params, this is not used anymore.

File: tools/ggplot2/ggplot_histogram.xml [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+        <conditional name="transformation">
+            <param name="transform" type="select" label="Advanced - log transformation">
+              <option value="none">Plot the data as it is</option>
+              <option value="log2">Log2(value) transform my data</option>
+              <option value="log2plus1">Log2(value+1) transform my data</option>
+              <option value="log10">Log10(value) transform my data</option>
+              <option value="log10plus1">Log10(value+1) transform my data</option>
+            </param>
+        </conditional>
+        <conditional name="distribution">
```
Comment (bgruening) [inline tools/ggplot2/ggplot_histogram.xml]: Any reason you are using a conditional here? You don't have a `when`.

File: tools/ggplot2/ggplot_histogram.xml [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+            <param name="plotdim" type="select" label="Advanced - output dimensions">
+                <option value="Default" selected="True">Default</option>
+                <option value="Defined">User deined output dimensions</option>
+            </param>
+            <when value="Default">
+                <!--Do nothing here -->
+            </when>
+            <when value="Defined">
+                <param name="woutputdim" size="4" type="float" value="7" label="width of output (inches)" />
+                <param name="houtputdim" size="4" type="float" value="7" label="height of output (inches)" />
```
Comment (bgruening) [inline tools/ggplot2/ggplot_histogram.xml]: remove size here as well ...

File: tools/ggplot2/ggplot_histogram.xml [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+            </when>
+        </conditional>
+        <conditional name="showlegend">
```
Comment (bgruening) [inline tools/ggplot2/ggplot_histogram.xml]: remove conditional

File: tools/ggplot2/ggplot_histogram.xml [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+            </param>
+        </conditional>
+    </inputs>
+    <outputs>
+        <!-- <data name="output1" format="png" from_work_dir="Rplot.png"/> -->
+        <data name="output1" format="pdf" from_work_dir="Rplot.pdf"/>
+    </outputs>
+    <tests>
+        <test>
+            <param name="input1" value="input.txt"/>
```
Comment (bgruening) [inline tools/ggplot2/ggplot_histogram.xml]: maybe adding some parameters, or an additional test with parameters?

File: tools/ggplot2/ggplot_histogram.xml [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+gene_A  10  15
+gene_B  8   12
+gene_C 10   15
+gene_D  6   9
+gene_E  9   13.5
+gene_F  8   12
+
+    ]]></help>
+    <citations>
+        <citation type="bibtex">
```
Comment (bgruening) [inline tools/ggplot2/ggplot_histogram.xml]: Either remove this or add your name :)

File: tools/ggplot2/ggplot_point.xml [hunk 9]
```diff
@@ -0,0 +1,390 @@
+<tool id="ggplot_point" name="Scatterplot w ggplot2" version="0.1.3">
+    <requirements>
+        <requirement type="package">r-getopt</requirement>
+        <requirement type="package">r-ggplot2</requirement>
+    </requirements>
+    <stdio>
```
Comment (bgruening) [inline tools/ggplot2/ggplot_point.xml]: Same as above.

File: tools/ggplot2/ggplot_point.xml [hunk 10]
```diff
@@ -0,0 +1,390 @@
+<tool id="ggplot_point" name="Scatterplot w ggplot2" version="0.1.3">
+    <requirements>
```
Comment (bgruening) [inline tools/ggplot2/ggplot_point.xml]: You might want to put the reuqirement tags into a macro? As well as the color table, the citations and everything else what is repeated?

**Comment by bgruening** [comment]
@MoHeydarian I added a WIP for the time being, please let me know if we can/should help here! Would be great to have these tools soon on Galaxy main.

**Comment by MoHeydarian** [comment]
Thanks, @bgruening. I'm in the midst of GCC prep (and preparing a manuscript), I'll get back on this as soon as I can.

**Comment by bgruening** [comment]
Thanks @bimbam23!

**Comment by bgruening** [comment]
superseded by https://github.com/galaxyproject/tools-iuc/pull/1430

**Pull request closed without merging**